tasmota/xsns_12_ads1115.ino

```diff
@@ -133,13 +138,17 @@ void Ads1115StartComparator(uint8_t channel, uint16_t mode)
   uint16_t config = mode |
                     ADS1115_REG_CONFIG_CQUE_NONE    | // Comparator enabled and asserts on 1 match
                     ADS1115_REG_CONFIG_CLAT_NONLAT  | // Non Latching mode
-                    ADS1115_REG_CONFIG_PGA_6_144V   | // ADC Input voltage range (Gain)
+                     Ads1115.range                  | // ADC Input voltage range (Gain)
                     ADS1115_REG_CONFIG_CPOL_ACTVLOW | // Alert/Rdy active low   (default val)
                     ADS1115_REG_CONFIG_CMODE_TRAD   | // Traditional comparator (default val)
                     ADS1115_REG_CONFIG_DR_6000SPS;    // 6000 samples per second
 
-  // Set single-ended input channel
-  config |= (ADS1115_REG_CONFIG_MUX_SINGLE_0 + (0x1000 * channel));
+  // Set single-ended or differential input channel
+  if (Ads1115.channels == ADS1115_SINGLE_CHANNELS) {
+    config |= (ADS1115_REG_CONFIG_MUX_SINGLE_0 + (0x1000 * channel));
+  } else {
+    config |= (ADS1115_REG_CONFIG_MUX_DIFF_0_1 + (0x1000 * channel));
```

Comment:
So in single ended mode, as before, the driver supports channels 0..3 which are AIN0 to AIN3.
In the differential mode that you added, only 2 channels, as expected. But those 2 channels are 
* 0 = AIN0-AIN1 and 
* 1 = AIN0-AIN3 

which makes AIN2 useless.
It would be more versatile to use:
* 0 = AIN0 - AIN1
* 3 = AIN2 - AIN3

Which can be easily achieved by using 
```C++
config |= (ADS1115_REG_CONFIG_MUX_DIFF_0_1 + (0x3000 * channel));
```

Reply:
You are quite right. I'll change it this way..

Reply:
Done, took two tries. I realy should no do a commit on a tablet.